src/VettedGate.sol

```diff
@@ -179,6 +245,27 @@ contract VettedGate is
         ACCOUNTING.setBondCurve(nodeOperatorId, curveId);
     }
 
+    /// @inheritdoc IVettedGate
+    function claimReferrerBondCurve(uint256 nodeOperatorId) external {
+        NodeOperator memory nodeOperator = MODULE.getNodeOperator(
+            nodeOperatorId
+        );
+        address nodeOperatorAddress = nodeOperator.extendedManagerPermissions
+            ? nodeOperator.managerAddress
+            : nodeOperator.rewardAddress;
+        if (nodeOperatorAddress != msg.sender) {
+            revert NotAllowedToClaim();
+        }
+
+        if (referralCounts[_seasonedAddress(msg.sender)] < referralsThreshold) {
+            revert NotEnoughReferrals();
+        }
```

Comment:
would suggest to have it in `_consumeReferrer()`.  At least to avoid calling `_seasonedAddress()` twice